internal/service/encryptionatrestprivateendpoint/state_transition.go

```diff
@@ -0,0 +1,70 @@
+package encryptionatrestprivateendpoint
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/retrystrategy"
+	"go.mongodb.org/atlas-sdk/v20240805001/admin"
+)
+
+func waitStateTransition(ctx context.Context, projectID, cloudProvider, endpointID string, client admin.EncryptionAtRestUsingCustomerKeyManagementApi) (*admin.EARPrivateEndpoint, error) {
+	return WaitStateTransitionWithMinTimeout(ctx, 1*time.Minute, projectID, cloudProvider, endpointID, client)
+}
+
+func WaitStateTransitionWithMinTimeout(ctx context.Context, minTimeout time.Duration, projectID, cloudProvider, endpointID string, client admin.EncryptionAtRestUsingCustomerKeyManagementApi) (*admin.EARPrivateEndpoint, error) {
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{retrystrategy.RetryStrategyInitiatingState},
+		Target:     []string{retrystrategy.RetryStrategyPendingAcceptanceState, retrystrategy.RetryStrategyActiveState, retrystrategy.RetryStrategyFailedState},
+		Refresh:    refreshFunc(ctx, projectID, cloudProvider, endpointID, client),
+		Timeout:    20 * time.Minute,
+		MinTimeout: minTimeout,
+		Delay:      0,
+	}
+
+	result, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if privateEndpointResp, ok := result.(*admin.EARPrivateEndpoint); ok && privateEndpointResp != nil {
+		return privateEndpointResp, nil
+	}
+	return nil, errors.New("did not obtain valid result when waiting for state transition")
+}
+
+func WaitDeleteStateTransition(ctx context.Context, projectID, cloudProvider, endpointID string, client admin.EncryptionAtRestUsingCustomerKeyManagementApi) error {
+	return WaitDeleteStateTransitionWithMinTimeout(ctx, 1*time.Minute, projectID, cloudProvider, endpointID, client)
+}
+
+func WaitDeleteStateTransitionWithMinTimeout(ctx context.Context, minTimeout time.Duration, projectID, cloudProvider, endpointID string, client admin.EncryptionAtRestUsingCustomerKeyManagementApi) error {
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{retrystrategy.RetryStrategyPendingAcceptanceState, retrystrategy.RetryStrategyActiveState, retrystrategy.RetryStrategyPendingRecreationState, retrystrategy.RetryStrategyFailedState},
```

Comment:
If new DELETING status is added this will need adjustment.

Reply:
if you plan to merge this right away, could we create a ticket for this?

Reply:
adjusted as part of this PR